### DISCUSSION TITLE
Help me

### DISCUSSION THREAD

**Discussion by madfortech** [laravel/framework#50297] [Q&A]
![Screenshot 02-28-2024 18 44 43](https://github.com/laravel/framework/assets/21261716/3d6a708d-5905-475c-ba1f-8adc44704b14)

![Screenshot 02-28-2024 18 17 43](https://github.com/laravel/framework/assets/21261716/32f82984-5682-4e5f-a3f2-9e8a98e0a2e4)
I have created a file. How do I test it? In laravel project. before i give pull request please help me
![Screenshot 02-28-2024 18 40 47](https://github.com/laravel/framework/assets/21261716/6eafb167-a03c-47e7-864f-ed862c03710b)

when i run composer install our project automatically created new file in my laravel project ?

**Comment by ghost** [comment]
Composer install creates the vendor directory and composer.lock file if not there already.

**Comment by ModulesSoft** [comment]
>  I have created a file. How do I test it? In laravel project. before i give pull request please help me !


You can use PHPUnit or Pest to start testing your code. 
Learn more here:
https://laravel.com/docs/testing

> when i run composer install our project automatically created new file in my laravel project ?

"A repository is a package source. It's a list of packages/versions. Composer will look in all your repositories to find the packages your project requires."
https://getcomposer.org/doc/05-repositories.md

Therefore, I think your composer.json specifies a local path repository for a package located in a directory named "framework" one level above the current project directory.

By default, Laravel's basic structure does not include a 'repositories' attribute in its composer.json file. 
See: https://github.com/laravel/framework/blob/10.x/composer.json
And when you run composer install it doesn't make any file other than composer.lock and the vendor directory, as @marius-mcp mentioned.
 
You may need to remove the extra lines there.
